File: crates/client/client-core/src/events.rs
use bytes::Bytes;
use cli_pocket_proto::{SessionId, StreamSeq, TerminalId, TerminalInfo};

#[derive(Debug, Clone)]
pub enum ClientEvent {
    Connecting,
    Connected {
        session_id: SessionId,
        server_label: Option<String>,
    },
    Disconnected {
        will_retry: bool,
        reason: String,
    },
    TerminalCreated(TerminalInfo),
    TerminalOutput {
        terminal_id: TerminalId,
        stream_seq: StreamSeq,
        bytes: Bytes,
    },
    Error(String),
}

File: webview/terminal/src/types/events.ts
import type { ExitInfo, StreamSeq, TerminalId, TerminalInfo } from "./frame";

export type ClientEvent =
  | { kind: "Connecting" }
  | { kind: "Connected"; session_id: string }
  | { kind: "Disconnected"; will_retry: boolean; reason: string }
  | { kind: "TerminalCreated"; info: TerminalInfo }
  | {
      kind: "TerminalOutput";
      terminal_id: TerminalId;
      stream_seq: StreamSeq;
      bytes_b64: string;
    }
  | { kind: "TerminalExited"; terminal_id: TerminalId; info: ExitInfo }
  | { kind: "Error"; message: string };
